room/views.py

```diff
@@ -1,37 +1,43 @@
-from math import floor
-from unicodedata import name
-from rest_framework.views import APIView
-
-from ..user.utils import admin_required
-from .serializers import RoomSerializer
-from room.models import Room
+from .serializers import HotelSerializer, RoomSerializer
 from rest_framework.response import Response
+from rest_framework.views import APIView
+from user.utils import admin_required
+from room.models import Room, Hotel
 from rest_framework import status
 
 # Create your views here.
 
 class RoomView(APIView):
     serializer_class = RoomSerializer
+    permission_classes = [admin_required]
     
     def get(self, request, id=None):
         user = Room.objects.all()
+        price = request.GET.get('price', None)
+        if price is not None:
+            user = user.filter(price=price)
```

Comment:
price should be a range, not an actual number

Reply:
fixed